cmd/fetchbranches/branch.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"regexp"
+	"strings"
+)
+
+// Branch is the structure providing specific parameter for the GetBranch execution
+type Branch struct {
+	Repo string
+	URL  string
+}
+
+// Valid tag names
+var tags = []string{"proxy", "shim", "tests", "hyperstart", "runtime"}
+
+// GetBranch retrieves the branches from the comment
+func GetBranch(comment string) []Branch {
+	var branches []Branch
+
+	for _, t := range tags {
+		// Obtaining full name (branch and url)
+
+		regex := regexp.MustCompile("branch_" + t + ":[[:space:]]*http[s]{0,1}:[[:graph:]]+")
+		index := regex.FindStringIndex(comment)
+		if len(index) == 0 {
+			continue
+		}
+
+		// Valid sintaxis
+		// i.e branch_test: https://xxxx.com/
+		tag := comment[index[0]:index[1]]
+
+		// Looking for the beggining of the URL (http)
+		i := strings.Index(tag, "http")
```

Comment:
This looks fine, but I'm wondering:

- If there is an easier way. I came across https://github.com/mvdan/xurls which might be worth a look.
- About validation. It wouldn't hurt to call:
  ```
  possibleURL := tag[i:]

  // ensure it really is a valid-looking URL
  _, err := url.Parse(possibleURL)
  if err != nil {
      fmt.Fprintf(os.Stderr, "WARNING: ignoring url: %v\n", possibleURL)
      continue
  }
  ```

@dlespiau - any thoughts on this?